includes/namespace.php

```diff
@@ -277,6 +279,83 @@ function deactivate( $network_wide ) {
 register_activation_hook( WEBSTORIES_PLUGIN_FILE, __NAMESPACE__ . '\activate' );
 register_deactivation_hook( WEBSTORIES_PLUGIN_FILE, __NAMESPACE__ . '\deactivate' );
 
+
+/**
+ * Append result of internal request to REST API for purpose of preloading data to be attached to a page.
+ * Expected to be called in the context of `array_reduce`.
+ *
+ * Like rest_preload_api_request() in core, but embeds links.
+ *
+ * @link  https://core.trac.wordpress.org/ticket/51636
+ *
+ * @since 1.2.0
+ *
+ * @see   \rest_preload_api_request
+ *
+ * @param array        $memo Reduce accumulator.
+ * @param string|array $path REST API path to preload.
+ *
+ * @return array Modified reduce accumulator.
+
+ */
+function rest_preload_api_request( $memo, $path ) {
+	// array_reduce() doesn't support passing an array in PHP 5.2,
+	// so we need to make sure we start with one.
+	if ( ! is_array( $memo ) ) {
+		$memo = [];
+	}
+
+	if ( empty( $path ) ) {
+		return $memo;
+	}
+
+	$method = 'GET';
+	if ( is_array( $path ) && 2 === count( $path ) ) {
+		$method = end( $path );
+		$path   = reset( $path );
+
+		if ( ! in_array( $method, [ 'GET', 'OPTIONS' ], true ) ) {
+			$method = 'GET';
+		}
+	}
+
+	$path_parts = wp_parse_url( $path );
+	if ( false === $path_parts ) {
+		return $memo;
+	}
+
+	$request = new WP_REST_Request( $method, $path_parts['path'] );
+	$embed   = false;
+	if ( ! empty( $path_parts['query'] ) ) {
+		$query_params = [];
+		parse_str( $path_parts['query'], $query_params );
+		$embed = isset( $query_params['_embed'] );
```

Comment:
```suggestion
		$embed = $query_params['_embed'];
```

This would be more accurate I think

Reply:
```suggestion
		$embed = isset( $query_params['_embed'] ) ? $query_params['_embed'] : false;
```
Remember it is a key in an array, so we need to check it exists first. I think that would be better.

Reply:
Right! Makes sense like that 👍